ReactWindows/ReactNative.Shared/Bridge/NativeModuleRegistry.cs

```diff
@@ -171,16 +171,13 @@ internal async Task NotifyReactInstanceInitializeAsync()
         /// Hook to notify modules that the <see cref="IReactInstance"/> has
         /// been disposed.
         /// </summary>
-        internal void NotifyReactInstanceDispose()
+        /// <returns>Awaitable task.</returns>
+        internal async Task NotifyReactInstanceDisposeAsync()
         {
             _reactContext.AssertOnNativeModulesQueueThread();
             using (Tracer.Trace(Tracer.TRACE_TAG_REACT_BRIDGE, "NativeModuleRegistry_NotifyReactInstanceDestroy").Start())
             {
-                foreach (var module in _moduleInstances.Values)
-                {
-                    Dispatch(module, module.OnReactInstanceDispose);
-                    module.ActionQueue?.Dispose();
-                }
+                await Task.WhenAll(_moduleInstances.Values.Select(DisposeModuleAsync));
```

Comment:
`return Task.WhenAll` + remove `async`?

Reply:
?

---
In reply to: [215817225](https://github.com/Microsoft/react-native-windows/pull/1958#discussion_r215817225) [](ancestors = 215817225)

Reply:
Nevermind - we "trace" the duration. Disregard.